worlds/witness/hints.py

```diff
@@ -747,11 +823,17 @@ def make_compact_hint_data(hint: WitnessWordedHint, local_player_number: int) ->
     location = hint.location
     area_amount = hint.area_amount
 
+    player_number = local_player_number
+    if location:
+        player_number = location.player
+    if player_number == local_player_number and hint.confusified_hint:
+        player_number = -1
```

Comment:
I think I saw you ask this question recently, but have you triple checked that it's safe to use `-1` as your own special value instead of player?